### DISCUSSION TITLE
Can we expect M3 Collector to be rolled out in the near future?

### DISCUSSION THREAD

**Discussion by Jugis** [m3db/m3#4164] [General]
Hey folks,

I was watching this video (watch?v=cHx4DuqINTM, start time: 36:00)from Uber Engineering about M3, and at the "in the future" section a seemingly not-yet-existing component is mentioned: The M3 collector, thats exactly what we would need. 

Can we expect this to be rolled out anytime soon? Has the development made progress on this department?

Thanks,
Mark

**Comment by Jugis** [comment]
Because the "traditional approach" levarages Prometheus remote/write APIs, basically its a stack on top of Prometheus. The issue is however is that we need to figure out how to distribute the load of the applications being monitored among the Prometheus instances upon which M3 could work. That would require some kind of sharding logic, failover etc, that we would not implement on our own. 

However IIRC Thanos resolves these issues, in case the M3 collector is not gonna be released any time soon, does any of you happen to know if M3DB is compatible with Thanos? Has anyone experimented with this?

**Reply by asafm** [reply]
M3 has a Coordinator component, which is almost stateless, and you can add more instances if you want to achieve high throughput. You can do load-balancing between them (up to a certain degree). So why M3 Collector helps here in any way? I actually don't understand why you say Thanos resolves this issue.
